include/zephyr/arch/cache.h

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2023 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * Public APIs for architectural cache controller drivers
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_CACHE_H_
+#define ZEPHYR_INCLUDE_ARCH_CACHE_H_
+
+/**
+ * @brief Cache Controller Interface
+ * @defgroup cache_arch_interface Cache Controller Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#if defined(CONFIG_ARM64)
```

Comment:
Might there be too many `#if deifned()` for different `arch` in the future?

Reply:
well, it's a common pattern in Zephyr for arch headers: https://github.com/zephyrproject-rtos/zephyr/blob/c4565ad5b6527dd2474067230f98108518324ab4/include/zephyr/arch/cpu.h#L14-L36

Reply:
Ah, right, forgot it's a header